test/test_transforms_task.py

```diff
@@ -713,3 +713,72 @@ def test_treeherder_defaults(run_transform, graph_config, kind, task_def, expect
     pprint(task_dict, indent=2)
 
     assert task_dict["task"].get("extra", {}).get("treeherder", {}) == expected_th
+
+
+def test_check_task_dependencies(graph_config):
+    params = FakeParameters(
+        {
+            "base_repository": "git@github.com://github.com/mozilla/example.git",
+            "build_date": 0,
+            "build_number": 1,
+            "head_repository": "git@github.com://github.com/mozilla/example.git",
+            "head_rev": "abcdef",
+            "head_ref": "default",
+            "level": "1",
+            "moz_build_date": 0,
+            "next_version": "1.0.1",
+            "owner": "some-owner",
+            "project": "some-project",
+            "pushlog_id": 1,
+            "repository_type": "git",
+            "target_tasks_method": "test_method",
+            "tasks_for": "github-pull-request",
+            "try_mode": None,
+            "version": "1.0.0",
+        },
+    )
+
+    transform_config = TransformConfig(
+        "check_task_dependencies",
+        str(here),
+        {},
+        params,
+        {},
+        graph_config,
+        write_artifacts=False,
+    )
+
+    task1 = {
+        "label": "task1",
+        "dependencies": ["dependency1", "dependency2"],
+        "soft-dependencies": ["dependency3"],
+        "if-dependencies": ["dependency4"],
+    }
+    task2 = {
+        "label": "task2",
+        "dependencies": ["dependency"] * 97,
+        "soft-dependencies": ["dependency"],
+        "if-dependencies": ["dependency"],
+    }
+
+    # Test tasks with less than 100 dependencies(task1) including exactly 99 dependencies(task2)
+    assert (
+        len(list(task.check_task_dependencies(transform_config, [task1, task2]))) == 2
```

Comment:
Improvement: Good call on testing several types of task! We can write this test in a more generic way with `@pytest.mark.parametrize`. For example: https://github.com/taskcluster/taskgraph/blob/a0dd5d37b31b4912969b87bd57c14462cb0a1868/test/test_transforms_task.py#L75

It would look like: 

```py
@pytest.mark.parametrize("number_of_dependencies, expectation", (
    (2, does_not_raise()),
    (97, does_not_raise()),
    (98, does_not_raise()),
    (99, pytest.raises(Exception)),
def test_check_task_dependencies(graph_config, number_of_dependencies, expectation):
```

More info in https://docs.pytest.org/en/7.1.x/example/parametrize.html?highlight=does_not_raise#parametrizing-conditional-raising

Let me know if you need more information 🙂

Reply:
I have updated and written the unit test "test_check_task_dependencies" in the generic way given above. Let me know if any more improvement is needed. Thank you!